providers/softlayer/src/test/java/org/jclouds/softlayer/features/NetworkApiLiveTest.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.softlayer.features;
+
+import com.google.common.collect.ImmutableList;
+import org.jclouds.softlayer.domain.Network;
+import org.jclouds.softlayer.domain.Subnet;
+import org.jclouds.softlayer.domain.Datacenter;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNull;
+
+/**
+ * Tests behavior of {@code NetworkApi}
+ */
+@Test(groups = "live")
+public class NetworkApiLiveTest extends BaseSoftLayerApiLiveTest {
```

Comment:
Unfortunately my account is not allowed to create networks so I couldn't run the `NetworkApiLiveTest` -- `org.jclouds.http.HttpResponseException: command: POST https://api.softlayer.com/rest/v3.1/SoftLayer_Network/createObject HTTP/1.1 failed with response: HTTP/1.1 500 Internal Server Error; content: [{"error":"Cloudsoft Corporation (#xxxxx) is not allowed to create networks.","code":"SoftLayer_Exception_Account_Limit"}]` so could you maybe paste here the results from your account, maybe? thanks!

Reply:
Hi Andrea, I believe that only certain type of accounts have access to the new SL+ features.
The following is a mvn test report results:

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
Running org.jclouds.softlayer.features.NetworkApiLiveTest
Configuring TestNG with: TestNG652Configurator
Starting test testCreateNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testCreateNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 1710ms
Test suite progress: tests succeeded: 1, failed: 0, skipped: 0.
Starting test testCreateSubnet(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testCreateSubnet(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 3401ms
Test suite progress: tests succeeded: 2, failed: 0, skipped: 0.
Starting test testGetName(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testGetName(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 1206ms
Test suite progress: tests succeeded: 3, failed: 0, skipped: 0.
Starting test testGetNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testGetNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 1325ms
Test suite progress: tests succeeded: 4, failed: 0, skipped: 0.
Starting test testGetNotes(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testGetNotes(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 1147ms
Test suite progress: tests succeeded: 5, failed: 0, skipped: 0.
Starting test testListNetworks(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testListNetworks(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 7443ms
Test suite progress: tests succeeded: 6, failed: 0, skipped: 0.
Starting test testEditNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testEditNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 2570ms
Test suite progress: tests succeeded: 7, failed: 0, skipped: 0.
Starting test testGetSubnets(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testGetSubnets(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 1108ms
Test suite progress: tests succeeded: 8, failed: 0, skipped: 0.
Starting test testDeleteSubnet(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testDeleteSubnet(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 2658ms
Test suite progress: tests succeeded: 9, failed: 0, skipped: 0.
Starting test testDeleteNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest)
[TestNG] Test testDeleteNetwork(org.jclouds.softlayer.features.NetworkApiLiveTest) succeeded: 11568ms
Test suite progress: tests succeeded: 10, failed: 0, skipped: 0.
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 53.068 sec - in org.jclouds.softlayer.features.NetworkApiLiveTest

Results :

Tests run: 10, Failures: 0, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:05 min
[INFO] Finished at: 2017-11-21T18:07:17+08:00
[INFO] Final Memory: 27M/330M
[INFO] ------------------------------------------------------------------------
```